src/EditorFeatures/CSharpTest/InlineDeclaration/CSharpInlineDeclarationTests.cs

```diff
@@ -0,0 +1,347 @@
+        if (int.TryParse(v, out i))
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestAvailableInOuterScopeIfNotWrittenOutside()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i = 0;
+        {
+            if (int.TryParse(v, out i))
+            {
+            }
+            i = 1;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingIfWrittenAfterInOuterScope()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i = 0;
+        {
+            if (int.TryParse(v, out i))
+            {
+            }
+        }
+        i = 1;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingIfWrittenBetweenInOuterScope()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i = 0;
+        {
+            i = 1;
+            if (int.TryParse(v, out i))
+            {
+            }
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingInNonOut()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i;
+        if (int.TryParse(v, i))
+        {
+        } 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingInField()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    [|int|] i;
+    void M()
+    {
+        if (int.TryParse(v, out this.i))
+        {
+        } 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingInField2()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    [|int|] i;
+    void M()
+    {
+        if (int.TryParse(v, out i))
+        {
+        } 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingInNonLocalStatement()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        foreach ([|int|] i in e)
+        {
+            if (int.TryParse(v, out i))
+            {
+            }
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task TestMissingInEmbeddedStatement()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M()
+    {
+        [|int|] i;
+        while (true)
+            if (int.TryParse(v, out i))
+            {
+            } 
+    }
+}");
```

Comment:
Yes.  You are correct.  This *should* work.  For expediency of the implementation it doesn't (since it's much easier to do all the analysis on the real block-syntaxes that exist.  I will make a note in the test that we would like this to work, and i will attempt to make it work if it's not a huge effort.